src/nifc/codegen.nim

```diff
@@ -82,6 +82,7 @@ type
     headerFile: seq[Token]
     generatedTypes: IntSet
     requestedSyms: HashSet[string]
+    toExtern: bool
```

Comment:
Instead of this solution, why not call `genVar` instead of `genStmt`?

Reply:
Sure, I will extract them into a `genVar` function to avoid duplication and pass `toExtern` by parameters